### DISCUSSION TITLE
Configuring a plugin: onedarkpro

### DISCUSSION THREAD

**Discussion by dpo** [LunarVim/LunarVim#2829] [General]
I installed the onedarkpro theme, but sadly, the theme disables the cursorline by default. I tried to configure it like so
```lua
lvim.colorscheme = "onedarkpro"

-- ...

lvim.plugins = {
  {
    "olimorris/onedarkpro.nvim",
    config = function()
      require("onedarkpro").setup({
        options = {
          cursorline = true,
        },
      }),
    end,
  },
}
```
Unfortunately, still no cursorline. The configuration of LunarVim is still a bit of a mystery. What am I doing wrong here?
Here is the onedarkpro default config: https://github.com/olimorris/onedarkpro.nvim#package-installation

Thanks!

**Comment by rebuilt** [comment]
Check to see if the cursorline is actually enabled  `:set cursorline?` .  The output of that command should be `cursorline`. 

Make sure termguicolors is enabled .   `:set termguicolors?` .  The output should be `termguicolors`

Enable with
```lua
vim.o.termguicolors = true
```

**Comment by dpo** [comment]
Thank you for your reply.

I have set
```lua
vim.opt.cursorline = true
vim.opt.termguicolors = true
```

`:set cursorline?` and `:set termguicolors?`  output `curcorline` and `termguicolors`, respectively. But still no cursorline.

**Comment by rebuilt** [comment]
The [readme](https://github.com/olimorris/onedarkpro.nvim#cursorline) states that, by default, the cursorline color is 'based' on the background color.   I'm not sure if this means they are the same color.  If they are the same color, you won't be able to see a difference.  I'd try setting the cursorline to yellow(#FFF000).

```lua
colors = {
  cursorline = "#FFF000" -- This is optional. The default cursorline color is based on the background
},
options = {
  cursorline = true
}
```

**Reply by dpo** [reply]
This had no effect when I tried. It's as if the options weren't taken into account. Did it work for you?

**Reply by rebuilt** [reply]
Yes,  it worked when I tested it on my instance of LunarVim.  Sometimes changes don't take effect because of a problem in the packer compiled file.   You can delete it from `~/.config/lvim/plugin` and run `:PackerSync` again to make sure it compiles ok.  

![onedarkpro](https://user-images.githubusercontent.com/16025007/180510793-99877d0d-be46-462d-b31e-05b248be4c3d.png)

**Comment by rebuilt** [comment]
You can also try setting it manually with 
```
:hi CursorLine guibg=yellow
```

**Comment by dpo** [comment]
Many thanks for your assistance! Setting `:hi CursorLine guibg=yellow` works for me too. I like to switch between a dark and light theme depending on the time of day, and while setting the CursorLine color manually for both types of theme is perfectly doable, it seems like OneDarkPro is either buggy or there is an incomaptiblity with LunarVim. Removing the Packer compiled file didn't solve the issue either. Modifying the source code of OneDarkPro to enable the cursor line by default worked, but that's obviously a hack.

I abandoned OneDarkPro and decided to install [OneDark](https://github.com/navarasu/onedark.nvim) instead. I set
```lua
lvim.plugins = {
  {
    "navarasu/onedark.nvim",
    config = function()
      require('onedark').setup {
        style = 'dark', -- Default theme style. Choose between 'dark', 'darker', 'cool', 'deep', 'warm', 'warmer' and 'light'
        transparent = false, -- Show/hide background
        term_colors = true, -- Change terminal color as per the selected theme style
        ending_tildes = false, -- Show the end-of-buffer tildes. By default they are hidden
        cmp_itemkind_reverse = false, -- reverse item kind highlights in cmp menu
        -- toggle theme style ---
        toggle_style_key = '<leader>ts', -- Default keybinding to toggle
        toggle_style_list = { 'darker', 'light' }, -- List of styles to toggle between
      }
    end,
  },
}
```
That just works. Switching between dark and light themes works out of the box, and, like OneDarkPro, is synced with the background color. 

As a side note, it could be nice to ship a theme that has both a dark and light variants with LunarVim.

And finally, a related question: since I've installed a new theme (OneDarkPro), LunarVim keeps pestering me about removing OneDarker each time I update the config file:

![Screen Shot 2022-07-22 at 14 10 03](https://user-images.githubusercontent.com/38760/180518198-658dc361-6a3e-4be4-b9fb-59b8e130fbc1.png)

Is there a way to instruct LunarVim to stop asking? I don't want to remove anything.

**Reply by rebuilt** [reply]
There's no way to disable the prompt for uninstalling.  At least not without a pull request to modify how we reload the user configuration file.  The prompt isn't something we've coded in.  It's a byproduct of the way packer behaves.   Packer is the plugin that manages the install and removal of other plugins for neovim.   You'd get the same behavior if you manually ran  the packer command `:PackerSync`.   

That said, it's really easy to just let packer install or uninstall the plugins you've defined in your config.   Are you afraid you won't be able to reinstall the plugins later or is there some other reason you want to avoid uninstalling plugins?

**Reply by dpo** [reply]
I find the message confusing because "OneDarker" is one of the standard plugins, not one I defined in my config. Why does it want to uninstall a standard plugin?

About OneDark, I take it back. The options defined in my config above are actually not taken into account. For example, the [default config](https://github.com/navarasu/onedark.nvim#default-configuration) states
```lua
toggle_style_list = {'dark', 'darker', 'cool', 'deep', 'warm', 'warmer', 'light'}, -- List of styles to toggle between
```
I changed that list above, but toggling toggles between all 7 styles, not just the two I defined.

Are you seeing an error in the way I pass options above? If not, it looks like those never make it to the plugin itself.
